doc/man3/SSL_CTX_set_quic_method.pod

```diff
@@ -0,0 +1,235 @@
+=pod
+
+=head1 NAME
+
+SSL_QUIC_METHOD,
+OSSL_ENCRYPTION_LEVEL,
+SSL_CTX_set_quic_method,
+SSL_set_quic_method,
+SSL_set_quic_transport_params,
+SSL_get_peer_quic_transport_params,
+SSL_quic_max_handshake_flight_len,
+SSL_quic_read_level,
+SSL_quic_write_level,
+SSL_provide_quic_data,
+SSL_process_quic_post_handshake,
+SSL_is_quic
+- QUIC support
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ typedef struct ssl_quic_method_st SSL_QUIC_METHOD;
+ typedef enum ssl_encryption_level_t OSSL_ENCRYPTION_LEVEL;
```

Comment:
Hmmm enum in a public API we tend not to like that.

Reply:
Unfortunately, it's how BoringSSL does it. For source compatibility, I could change this to be a `typedef int OSSL_ENCRYPTION_LEVEL` and `#define` the enumerated values.

Reply:
Actually... I can't swap it as the QUIC stack may use `enum ssl_encryption_t` directly.

Reply:
`OSSL_HANDSHAKE_STATE` is an enum. along with `SSL_CT_VALIDATION_...`, but yeah, this is an input type...